src/district/ui/graphql/events.cljs

```diff
@@ -10,16 +10,30 @@
     [district.ui.graphql.queries :as queries]
     [district.ui.graphql.utils :as utils]
     [graphql-query.core :refer [graphql-query]]
-    [re-frame.core :refer [reg-event-fx trim-v]]))
+    [re-frame.core :refer [reg-event-fx trim-v]]
+    [goog.object :as gobj]))
 
 (def interceptors [trim-v])
 
+(defn add-token-fetcher-middleware [data next]
+  (when-let [token (::authorization-token @re-frame.db/app-db)]
+    (when-not (.. data -options -headers)
+      (set! (.. data -options -headers) #js {}))
+    (gobj/set (.. data -options -headers) "authorization" (str "Bearer " token)))
+  (next))
+
+(reg-event-db
+ ::set-authorization-token
+ (fn [db [_ token]]
+   (assoc db ::authorization-token token)))
```

Comment:
To correctly follow our patterns, the correct way would be that there's function for this in `district.ui.graphql.queries`, which is using `:district.ui.graphql` as root db key. + added documentation
+ ideally little test for this, if it's not too complicated